junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/FilePosition.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine.discovery;
+
+import static org.apiguardian.api.API.Status.STABLE;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.commons.util.Preconditions;
+import org.junit.platform.commons.util.StringUtils;
+import org.junit.platform.commons.util.ToStringBuilder;
+
+/**
+ * Position inside a file represented by {@linkplain #getLine line} and
+ * {@linkplain #getColumn column} numbers.
+ *
+ * @since 1.0
+ */
+@API(status = STABLE, since = "1.0")
```

Comment:
Not particularly sure about the `since` argument here, and/or whether to declare this as `STABLE` already, given that we likely want to change the constructors from `protected` to `private`.

Reply:
You can also apply the annotation to specific methods to mark them as internal. The most specific annotation applies.